syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

// The greeting service definition.
service HiHo {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}

  // Send Bye
  rpc SayBye (ByeRequest) returns (ByeReply) {}
}

message HelloRequest {
  string name = 1;
}

message HelloReply {
  string message = 1;
}

message ByeRequest {
  string name = 1;
}

message ByeReply {
  string message = 1;
}

// regenerate using below command
// python3 -m grpc_tools.protoc -I. --python_out=. --pyi_out=. --grpc_python_out=. helloworld.proto